kernel/mmu.c

```diff
@@ -380,7 +380,12 @@ static void free_page_frame_list_put(struct z_page_frame *pf)
 {
 	PF_ASSERT(pf, z_page_frame_is_available(pf),
 		 "unavailable page put on free list");
-	sys_slist_append(&free_page_frame_list, &pf->node);
+	/* TODO: Check why this assertion hits */
+	/*PF_ASSERT(pf, IS_PTR_ALIGNED(pf, void*), "unaligned pf pointer");*/
```

Comment:
@dcpleung @andyross any idea why we are hitting this assert if I uncomment it?

Reply:
I'm not the expert on paging, but per the comment at the top of the struct in kernel/include/mmu.h it's intentionally misaligned for size optimization.  z_page_frame is a 5- or 9-byte struct (one pointer and a flags byte) and the intent seems to be to pack them all together, alignment be damned.

Reply:
OK, I will leave out the assertion then. Thanks!